build.sbt

```diff
@@ -177,27 +167,16 @@ lazy val sharedSettings = MimaSettings.settings ++ scalaVersionSettings ++ Seq(
   Compile / console / scalacOptions ~= {_.filterNot("-Ywarn-unused-import" == _)},
   Test / console / scalacOptions := (Compile / console / scalacOptions).value,
 
-  Compile / doc / sources := {
-    val old = (Compile / doc / sources).value
-    if (isDotty.value)
-      Seq()
-    else
-      old
-  },
-
   // don't use fatal warnings in tests
   Test / scalacOptions ~= (_ filterNot (_ == "-Xfatal-warnings")),
 
   mimaReportSignatureProblems := true,
   mimaPreviousArtifacts := {
-    // TODO: re-enable MiMa for Dotty once there is a final version
-    if (isDotty.value) Set()
-    else Set(
-      "1.14.3",
-      "1.15.1",
-      "1.15.2",
-      "1.15.3"
-    ).map(v => "org.scalacheck" %%% "scalacheck" % v)
+    // TODO: re-enable MiMa for Scala 3 once there is a final version
+    CrossVersion.partialVersion(scalaVersion.value) match {
+      case Some((3, _)) => Set()
+      case _            => Set("org.scalacheck" %%% "scalacheck" % "1.15.3")
```

Comment:
Why did you remove the other versions? I specifically added those in #761 to avoid any bloopers like we had before.

Reply:
I wasn't sure of the purpose.  We have been bitten by compatibility issues, but I recall those are more often from 1) upstream changes in MiMa, 2) things sneak in with version-specific code, and/or 3) because bootstrapping major Scala/Dotty release candidates or finals means ScalaCheck doesn't have a previous artifact published for MiMa to compare.